src/connection.rs

```diff
@@ -627,11 +625,13 @@ where
         self.sender.extend_interfaces(&interfaces, &to_add).await?;
 
         if let Some(retention) = self.store.get_retention() {
+            // TODO here stuff goes wrong if possible try to avoid using the boxed future
             let res = retention
                 .delete_interface_many(
-                    to_add
+                    &(to_add
                         .values()
-                        .filter_map(|v| v.is_major_change().then_some(v.interface_name())),
+                        .filter_map(|v| v.is_major_change().then_some(v.interface_name()))
+                        .collect::<Vec<_>>()),
```

Comment:
you can directly use `collect_vec()` from the (already imported) itertools crate instead of `collect::<Vec<_>>())`